### PR TITLE
docs: Remove api namespace.

### PR DESCRIPTION
**Pull request by mpreiner**

### CODE REVIEW

File: docs/api/cpp/cpp.rst [hunk 1]
```diff
@@ -3,7 +3,7 @@ C++ API
 
 The C++ API is the primary interface for cvc5 and exposes the full functionality of cvc5.
 The :doc:`quickstart guide <quickstart>` gives a short introduction, while the following class hierarchy of the ``cvc5::api`` namespace provides more details on the individual classes.
```
Comment (aniemetz) [inline docs/api/cpp/cpp.rst:5]:
```suggestion
The :doc:`quickstart guide <quickstart>` gives a short introduction, while the following class hierarchy of the ``cvc5`` namespace provides more details on the individual classes.
```

File: docs/api/cpp/cpp.rst [hunk 2]
```diff
@@ -35,15 +35,15 @@ Class hierarchy
 
 ``namespace cvc5::api {``
```
Comment (aniemetz) [inline docs/api/cpp/cpp.rst:36]:
```suggestion
``namespace cvc5 {``
```

File: docs/api/cpp/roundingmode.rst [hunk 3]
```diff
@@ -1,10 +1,10 @@
 RoundingMode
 ============
 
-.. doxygenenum:: cvc5::api::RoundingMode
+.. doxygenenum:: cvc5::RoundingMode
```
Comment (aniemetz) [inline docs/api/cpp/roundingmode.rst:4]: This generates a warning (cannot find enum "cvc5::RoundingMode"): https://cvc5.github.io/docs-ci/docs-pr8455/api/cpp/roundingmode.html

File: docs/api/cpp/roundingmode.rst [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
     :project: cvc5
 
-.. doxygenstruct:: std::hash< cvc5::api::RoundingMode >
+.. doxygenstruct:: std::hash< cvc5::RoundingMode >
```
Comment (aniemetz) [inline docs/api/cpp/roundingmode.rst:7]: Ditto

**Review by aniemetz — COMMENTED** [review]
In the theory references, the tables are not rendered anymore. You have to make sure that everything is aligned, else this will fail.

File: docs/api/cpp/roundingmode.rst (hunk 3 above)
Comment (mpreiner) [reply docs/api/cpp/roundingmode.rst:4]: I think that's due to https://github.com/cvc5/cvc5/pull/8427, I'll fix it in this PR.

**Review by aniemetz — APPROVED** [review]

**Pull request merged**